app/src/main/java/uk/gov/android/securestore/authentication/UserAuthenticator.kt

```diff
@@ -0,0 +1,61 @@
+package uk.gov.android.securestore.authentication
+
+import androidx.biometric.BiometricManager.Authenticators.BIOMETRIC_STRONG
+import androidx.biometric.BiometricManager.Authenticators.DEVICE_CREDENTIAL
+import androidx.biometric.BiometricPrompt
+import androidx.fragment.app.FragmentActivity
+import uk.gov.android.securestore.AccessControlLevel
+
+class UserAuthenticator(
+    private val context: FragmentActivity
+) : Authenticator {
+    override fun authenticate(
+        accessControlLevel: AccessControlLevel,
+        configuration: AuthenticatorPromptConfiguration,
+        handler: AuthenticatorCallbackHandler
+    ) {
+        val promptInfoBuilder = BiometricPrompt.PromptInfo.Builder()
+            .setTitle(configuration.title)
+            .setSubtitle(configuration.subTitle)
+            .setDescription(configuration.description)
+            .setAllowedAuthenticators(getRequireAuthenticators(accessControlLevel))
+
+        val biometricPrompt = BiometricPrompt(
+            context,
+            object : BiometricPrompt.AuthenticationCallback() {
+                override fun onAuthenticationError(
+                    errorCode: Int,
+                    errString: CharSequence
+                ) {
+                    super.onAuthenticationError(errorCode, errString)
+                    handler.onError(errorCode, errString)
+                }
+
+                override fun onAuthenticationSucceeded(
+                    result: BiometricPrompt.AuthenticationResult
+                ) {
+                    super.onAuthenticationSucceeded(result)
+                    handler.onSuccess()
+                }
+
+                override fun onAuthenticationFailed() {
+                    super.onAuthenticationFailed()
+                    handler.onFailure()
+                }
+            }
```

Comment:
Can we move this anonymous implementation somewhere please? We've got a lot of refactoring issues within the ID Check app due to this kind of pattern sharing state with other anonymous implementations so wouldn't want you to also have such hindrances.

Reply:
Yes good spot, have made the `AuthenticatorCallbackHandler` extend the `BiometricPrompt.AuthenticationCallback` so now we can simply pass the handler in